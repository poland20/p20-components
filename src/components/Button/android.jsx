// @flow

import * as React from 'react';
import { StyleSheet } from 'react-native';
import { Button } from 'native-base';

const style = StyleSheet.create({
  button: {
    backgroundColor: '#E2445C',
    paddingLeft: 15,
    paddingRight: 15,
  },
});

export default (props: any) => <Button {...props} style={[style.button, props.style]} />;
