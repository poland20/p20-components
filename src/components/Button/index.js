// @flow
import glamor from 'glamor';
import glamorous from 'glamorous';
import { colors } from '../variables';
import typography from '../typography';

const { rhythm } = typography;

const themes = {
  primary: {
    foreground: colors.white,
    background: colors.primary,
  },
};

function theme(themeName = 'primary', hollow = false) {
  const { foreground, background } = themes[themeName];
  if (hollow) {
    return {
      padding: `calc(${rhythm(0.5)} - 2px) calc(${rhythm(0.5)} - 2px)`,
      borderWidth: 2,
      borderStyle: 'solid',
      borderColor: background,
      color: background,
      backgroundColor: 'transparent',
      boxShadow: 'none',
    };
  }
  return {
    color: foreground,
    backgroundColor: background,
    ':hover': {
      backgroundColor: background.darken(0.1),
    },
  };
}

const style = {
  WebkitAppearance: 'none',
  borderRadius: 1,
  display: 'inline-block',
  padding: `${rhythm(0.5)} ${rhythm(1)}`,
  margin: `${rhythm(0.5)} 0`,
  position: 'relative',
  textAlign: 'center',
  verticalAlign: 'middle',
  touchAction: 'manipulation',
  textDecoration: 'none',
  whiteSpace: 'nowrap',
  border: 'none',
  cursor: 'pointer',
  boxShadow: '0px 2px 7px 0px rgba(1, 1, 1, 0.18)',
};

export const Button = glamorous.button(style, ({
  kind = 'primary',
  hollow = false,
}) => theme(kind, hollow));

export const NavButton = glamorous.a(style, ({
  kind = 'primary',
  hollow = false,
}) => theme(kind, hollow));

export default Button;
