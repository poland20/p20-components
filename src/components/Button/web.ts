import styled, { css } from 'react-emotion';
import { colors } from '../variables';
import typography from '../typography';

const { rhythm } = typography;

const themes = {
  primary: {
    foreground: colors.white,
    background: colors.primary,
  },
};

interface Props {
  // theme: 'primary';
  hollow?: boolean;
}

const theme = (props: Props) => {
  const { foreground, background } = themes['primary'];
  if (props.hollow) {
    return css({
      padding: `calc(${rhythm(0.5)} - 2px) calc(${rhythm(0.5)} - 2px)`,
      borderWidth: '2px',
      borderStyle: 'solid',
      borderColor: `${background}`,
      color: `${background}`,
      backgroundColor: 'transparent',
      boxShadow: 'none',
    });
  }

  return css({
    color: `${foreground}`,
    backgroundColor: `${background}`,
    '&:hover': {
      backgroundColor: `${background.darken(0.1)}`,
    },
  });
};

const style = css({
  WebkitAppearance: 'none',
  borderRadius: 1,
  display: 'inline-block',
  padding: `${rhythm(0.5)} ${rhythm(1)}`,
  margin: `${rhythm(0.5)} 0`,
  position: 'relative',
  textAlign: 'center',
  verticalAlign: 'middle',
  touchAction: 'manipulation',
  textDecoration: 'none',
  whiteSpace: 'nowrap',
  border: 'none',
  cursor: 'pointer',
  boxShadow: '0px 2px 7px 0px rgba(1, 1, 1, 0.18)',
});

export const Button = styled('button')(style, theme);

export const NavButton = styled('a')(style, theme);

export default Button;
