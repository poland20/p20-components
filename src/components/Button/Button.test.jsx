import React from 'react';
import Renderer from 'react-test-renderer';
import Button from '.';

it('renders correctly', () => {
  const tree = Renderer.create(<Button>Click me</Button>);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders hollow style correctly', () => {
  const tree = Renderer.create(<Button hollow>Click me</Button>);

  expect(tree.toJSON()).toMatchSnapshot();
});
