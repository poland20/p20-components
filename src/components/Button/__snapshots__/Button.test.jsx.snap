// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="css-s4f24e"
>
  Click me
</button>
`;

exports[`renders hollow style correctly 1`] = `
<button
  className="css-swf53u"
>
  Click me
</button>
`;
